### PR TITLE
[NRLF-434] Point Firehose at Splunk

### PR DESCRIPTION
**Pull request by jaklinger**

### CODE REVIEW

**Review by brianbayes — APPROVED** [review]

File: terraform/infrastructure/firehose.tf [hunk 1]
```diff
@@ -1,9 +1,15 @@
-
 module "firehose__processor" {
   source             = "./modules/firehose"
   prefix             = local.prefix
   region             = local.region
   layers             = [module.lambda-utils.layer_arn, module.nrlf.layer_arn, module.third_party.layer_arn]
   environment        = local.environment
   cloudwatch_kms_arn = module.kms__cloudwatch.kms_arn
+  splunk_index       = "aws_recordlocator_${contains(local.persistent_environments, local.environment) ? replace(local.environment, "-", "") : "dev"}"
```
Comment (jaklinger) [inline terraform/infrastructure/firehose.tf:8]: try to move this to `locals`

File: terraform/infrastructure/modules/firehose/kinesis.tf [hunk 2]
```diff
@@ -1,39 +1,106 @@
+
+
 resource "aws_kinesis_firehose_delivery_stream" "firehose" {
-  name        = "${var.prefix}-cloudwatch-to-splunk"
-  destination = "extended_s3"
+  name        = "${var.prefix}-cloudwatch--delivery-stream"
```
Comment (jaklinger) [inline terraform/infrastructure/modules/firehose/kinesis.tf:4]:
```
  name        = "${var.prefix}--cloudwatch-delivery-stream"
```

File: terraform/infrastructure/modules/firehose/kinesis.tf [hunk 3]
```diff
@@ -1,39 +1,106 @@
...
+  destination = var.destination
+  dynamic "splunk_configuration" {
+    # "splunk_configuration" is only enabled if the below secret has been set by hand in the AWS Console
+    #
+    # NB: the for_each will not run in the case that 'secret_string' has not been set in the AWS Console, and
+    # therefore this block will be omitted for default deployments (e.g. developer and CI workspaces).
+    for_each = { for item in data.aws_secretsmanager_secret_version.splunk_configuration.*.secret_string : "dummyKey" => jsondecode(item) }
```
Comment (jaklinger) [inline terraform/infrastructure/modules/firehose/kinesis.tf:11]: explanatory comment for "dummyKey"

**Comment by jaklinger** [comment]
source = "lambda-name"

**Pull request closed without merging**